### PR TITLE
Remove dead code from `TokenService`

### PR DESCRIPTION
**Pull request by n1v0lg**
This PR removes code marked as unused or redundant in `TokenService`.
More involved code removal to come.

Relates #87729

### CODE REVIEW

File: x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/TokenService.java [hunk 1]
```diff
@@ -1040,13 +1014,8 @@ private void findTokenFromRefreshToken(String refreshToken, Iterator<TimeValue>
...
                     listener.onResponse(SearchHits.EMPTY_WITH_TOTAL_HITS);
                 } else {
-                    // TODO Remove this conditional after backporting to 7.x
-                    if (refreshTokenVersion.onOrAfter(VERSION_HASHED_TOKENS)) {
-                        final String hashedRefreshToken = hashTokenString(unencodedRefreshToken);
-                        findTokenFromRefreshToken(hashedRefreshToken, securityTokensIndex, backoff, listener);
-                    } else {
-                        findTokenFromRefreshToken(unencodedRefreshToken, securityTokensIndex, backoff, listener);
-                    }
+                    final String hashedRefreshToken = hashTokenString(unencodedRefreshToken);
```
Comment (n1v0lg) [inline x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/TokenService.java:1017]: We've done the backport here: https://github.com/elastic/elasticsearch/pull/42220

File: x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/TokenService.java [hunk 2]
```diff
@@ -570,13 +549,8 @@ void decodeToken(String token, ActionListener<UserToken> listener) {
...
                 }
                 final String accessToken = in.readString();
-                // TODO Remove this conditional after backporting to 7.x
-                if (version.onOrAfter(VERSION_HASHED_TOKENS)) {
-                    final String userTokenId = hashTokenString(accessToken);
-                    getUserTokenFromId(userTokenId, version, listener);
-                } else {
-                    getUserTokenFromId(accessToken, version, listener);
-                }
+                final String userTokenId = hashTokenString(accessToken);
```
Comment (n1v0lg) [inline x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/TokenService.java:552]: We've done the backport here: #42220

**Review by tvernum — APPROVED** [review]
LGTM

**Comment by n1v0lg** [comment]
@elasticmachine update branch

**Pull request merged**